agent/credentials/providers/credentials_filename_linux.go

```diff
@@ -0,0 +1,22 @@
+//go:build linux
```

Comment:
Unfortunately, you really need to have both at the moment:

`// +build linux`

Reply:
Actually this ends in `_linux` you dont need it at all`

Reply:
I agree but I followed the convention followed across in this repo.

Reply:
We really need to have either both, or none. But half way is not good. Even if it is in the repo